portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Handles serving and creating comments through GET and POST respectively. 
+ * All requests must be accompanied by a projectId query-param, as comments are many-to-one to projects.
+ * Only comments for the requested project will be served.
+ */
+@WebServlet("/comments")
+public class CommentsServlet extends HttpServlet {
+
+  /**
+   * Serve a JSON array of all comments for the specified projectId.
+   * TODO: Add error response if no projectId is provided.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String projectId = request.getParameter("projectId");
+
+    /* 
+     * Get comments from datastore. 
+     * Filter to only show comments for this project.
+     */
```

Comment:
Just a note here: put more comments on the function, less inside.  Comments should let readers know why something unexpected is there, or call out something important.  Otherwise the code should let the user know what's happening.

In this case, the comment can be stripped down to:
```
// Filter to only show comments for this project.
```
and the Load query comment can go away.